api/models/user.js

```diff
@@ -174,10 +179,27 @@ const createOrFindUser = (
         console.error(err);
         return new Error(`No user found for id ${user.id}.`);
       }
+
+      // if no user exist check if a new one is contributor
+      if (providerMethod === 'githubProviderId') {
+        return storeUserWithContributionInfo(user);
+      }
       return storeUser(user);
     });
 };
 
+const storeUserWithContributionInfo = user =>
```

Comment:
just thinking out loud.
Maybe `storeUserWithContributionInfo` should be smt like `addContributionInfoToUser`. And look like this:
```Javascript
const addContributionInfoToUser = user =>
  isContributor(user.githubUsername, user.githubProviderId)
    .then(isContributor => {
      user.isContributor = isContributor;
      console.log(
        'Set isContributor for user: ',
        isContributor
      );
      return user;
    })
```
And the caller should handle the `storeUser(user);` and `.catch(err => storeUser(user));`?
```Javascript
if (providerMethod === 'githubProviderId') {
  return storeUserWithContributionInfo(user)
    .then(storeUser)
    .catch(err => storeUser(user);
}
```

This way we would handle the `storeUser` calls in `createOrFindUser` and not add this sideeffect to `storeUserWithContributionInfo`.

Reply:
Great catch @littleStudent, thats better to move that ```storeUser``` function outside because  ```addContributorInfoToUser``` should not know about persisting and it would be more reusable... thanks!